docker/Dockerfile.thumbv7m-none-eabi

```diff
@@ -10,15 +10,26 @@ RUN /cmake.sh
 COPY xargo.sh /
 RUN /xargo.sh
 
-COPY qemu.sh /
 RUN apt-get update && apt-get install --assume-yes --no-install-recommends \
     gcc-arm-none-eabi \
-    libnewlib-arm-none-eabi && \
-    /qemu.sh arm
+    libnewlib-arm-none-eabi \
+    libstdc++-arm-none-eabi-newlib
+
+COPY qemu.sh /
+RUN /qemu.sh arm
+
+COPY toolchain.cmake /opt/toolchain.cmake
+ENV CMAKE_TOOLCHAIN_FILE=/opt/toolchain.cmake
+ENV CROSS_CMAKE_OBJECT_FLAGS="-ffunction-sections -fdata-sections -mthumb -march=armv7-m"
 
 ENV CROSS_TOOLCHAIN_PREFIX=arm-none-eabi-
+ENV CROSS_SYSROOT="/usr/lib/arm-none-eabi"
```

Comment:
The THUMB newlib targets include their own binutils and include headers under this directory, so it's effectively a sysroot for most intents and purposes.